rest_framework_json_api/serializers.py

```diff
@@ -152,8 +153,51 @@ def validate_path(serializer_class, field_path, path):
         super(IncludedResourcesValidationMixin, self).__init__(*args, **kwargs)
 
 
+class LazySerializersDict(Mapping):
+    def __init__(self, klass, serializers):
+        self.klass = klass
+        self.serializers = serializers
+
+    def __getitem__(self, key):
+        value = self.serializers[key]
+        if not isinstance(value, type):
+            if value == "self":
+                value = self.klass
+            else:
+                value = import_class_from_dotted_path(value)
+            self.serializers[key] = value
+
+        return value
+
+    def __iter__(self):
+        return iter(self.serializers)
+
+    def __len__(self):
+        return len(self.serializers)
+
+    def __repr__(self):
+        return dict.__repr__(self.serializers)
+
+
 class SerializerMetaclass(SerializerMetaclass):
-    pass
+    def __new__(cls, name, bases, attrs):
+        klass = super().__new__(cls, name, bases, attrs)
+
+        if attrs.get("included_serializers", None):
```

Comment:
a bit odd to access `attrs` again after it already has been used to create above klass.

After the API change that `included_serializers` is always set (see other inline comment), this could be accessed as `klass.included_serializers`. To make it clearer it might also be better to rename `klass` to `serializer`.